storyboarding/scripting/compound_commands.rst

```diff
@@ -12,6 +12,405 @@ Loop
+  attention:: If you are planning to use Passing and Failing triggers in a map for either Catch the Beat or mania, go no further because these are literally not implemented.
+  There is full support for standard and partial support for Taiko (only Don/Kat, no drumrolls/spinners). Keep this in mind before making big plans with this type of trigger.
+
+As you might know already from reading the chapter about layers and objects, osu! supports 4 layers of which 2 are `Pass` and `Fail` that are displayed in accordance to the current game state.
+There are 2 trigger conditions corresponding with these layers called `Passing` and `Failing`:
+
+  - Passing occurs when the game state is changing from Fail to Pass
+  - Failing occurs when the game state is changing from Pass to Fail
+
+Now there is one major hiccup in applying this to storyboards and that is that Pass and Fail-Layers work differently in every game mode.
+
+In osu!standard the game state can only change at the end of a combo. If the last note of the combo gets you a Geki judgement the game state will be `Pass`, otherwise it will be `Fail`.
+
+In osu!taiko the game state can change on each note. If the last note was missed, the game state is `Fail`, otherwise it is `Pass`. It should also be noted that drumrolls (except for the head? unconfirmed) and spinners don't count as notes.
+
+In osu!mania and Catch the Beat the game state can only change on breaks and as we learned earlier, triggers always need to correspond to a gameobject. This means that `Passing` and `Failing` triggers basically don't exist for these gamemodes.
+
+
+Example: Change of Gamestate
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This example is geared towards osu!standard as it is the only gamemode that has a consistent implementation for these trigger conditions.
+
+It is rather advanced too but it would be boring otherwise, wouldn't it?
+
+
+The narrative
+.............
+
+Aoba slept in and got the late train! Her only chance to get to work in time is running from the train station to the Eagle Jump office.
+
+But...Aoba is clumsy. If she doesn't take care she will trip time and time again and not make it. It is up to the player to support Aoba in running.
+
+The plan
+........
+
+On the Background layer we will put a picture of a street that is sidescrolling.
+
+In Pass-state there will be an animation of Aoba running.
+
+In Fail-state Aoba will lie on the ground after having tripped.
+
+
+On triggering `Failing`, Aoba will be tripping.
+
+On triggering `Passing`, Aoba will get up from the ground.
+
+The implementation
+..................
+
+For the sidescrolling street we will take it easy as a start. After a quick google search a 2.5D animation of scrolling buildings turns up.
+
+Splitting that into frames, renaming the individual pictures to use as an animation (I used a script for this because it has about 250 files) and we're ready to go.
+
+.. image:: img/compound_commands/streetscroll.gif
```

Comment:
Man, this would make a cool effect. 👍

Reply:
Wikipedia article covering 2.5D, lmao
I agree.